### ISSUE TITLE
neurolucida.read(...) fails on some NeuroLucida files.

### ISSUE THREAD

**Issue by christian-oreilly** [BlueBrain/NeuroM#531]
I've been using the neurolucida.read(...) function which works fine on some of my files (from one collaborator) but not on others (from a different collaborator). The files that work have the format : 

```
("CellBody"
  (Color Blue)
  (CellBody)
  (    0.48    12.38     0.63     0.15)  ;  5, 1
  (   -0.32    12.70     0.63     0.15)  ;  5, 2
  (   -1.34    12.95     0.63     0.15)  ;  5, 3
  (   -2.30    12.92     0.81     0.15)  ;  5, 4
...
```

and those that don't have the format 

```
(Sections S1 "V2 section" 0 25 25)
(SSM "V2 section" 1)
(ImageCoords)

(""
  (Color White)
  (CellBody)
  (    3.34     0.81     0.00     0.09 S1)  ;  1, 1
  (    5.20     0.61     0.00     0.09 S1)  ;  1, 2
  (    7.89     1.61    -2.10     0.09 S1)  ;  1, 3
  (   10.77     3.03    -4.00     0.09 S1)  ;  1, 4
...
```
The key difference that makes them fail is the presence of the "S1" in the tuple  (    3.34     0.81     0.00     0.09 S1). If I remove those, then the reader works. With the "S1", the function _extract_section(section) always return an empty list.

**Comment by mgeplf** [comment]
We don't have a spec for the neurolucida files, so we have to work with data that people have sent us.  Does this have a different version string at the top of the file compared to:
`;       V3 text file written for MicroBrightField products.`

Also, does the file load correctly in NEURON?

**Comment by christian-oreilly** [comment]
Both samples start with 
`;	V3 text file written for MicroBrightField products.`

However, looking at these files with Elisabetta, we concluded that the `SX` identifies the slice. Let me give another example that is more clear on that:

```
;	V3 text file written for MicroBrightField products.
(Sections S1 "V2 section" 0 25 25
 S2 "V2 section" -25 25 25
 S3 "V2 section" 25 25 25
 S4 "V2 section" 50 25 25
 S5 "V2 section" -50 25 25
) ; End of Sections
(ImageCoords)

( (Color RGB (128, 255, 0))
  (Axon)
  (   24.65     9.03    -9.70     1.30 S1)  ; Root
  (   25.48    10.45   -11.50     1.30 S1)  ; 1, R
  (   25.48    12.06   -12.80     1.30 S1)  ; 2
  (   25.58    13.27   -12.80     1.11 S1)  ; 3
  (   25.85    15.39   -12.80     0.93 S1)  ; 4
  (   25.58    17.21   -12.80     0.74 S1)  ; 5

[...]

( (Color RGB (255, 128, 64))
  (Dendrite)
  (   11.38   -19.56    -5.12     3.90 S3)  ; Root
  (    9.71   -20.67    -4.42     3.71 S3)  ; 1, R
  (    7.76   -21.38    -4.02     3.16 S3)  ; 2
  (    6.18   -21.78    -2.02     2.78 S3)  ; 3
  (    4.79   -22.59    -2.02     2.60 S3)  ; 4
  (    3.31   -23.40    -0.42     2.60 S3)  ; 5
  (    1.45   -24.10    -0.02     2.60 S3)  ; 6
  (   -1.89   -24.81    -0.02     2.60 S3)  ; 7
  (   -4.50   -24.81    -0.02     2.23 S3)  ; 8
  (   -7.07   -25.73     1.78     2.04 S3)  ; 9
  (   -8.51   -26.80     1.78     1.86 S3)  ; 10
  (  -11.85   -26.93     1.78     1.86 S3)  ; 11
  (
    (  -12.81   -26.07     1.78     1.49 S3)  ; 1, R-1
    (
      (  -13.77   -25.30     0.68     1.11 S2)  ; 1, R-1-1
      (  -15.55   -24.97     0.68     1.11 S2)  ; 2
      (  -17.98   -24.76     0.68     0.93 S2)  ; 3
      (  -19.27   -23.60     0.68     0.93 S2)  ; 4
      (
        (  -21.00   -23.32    -0.92     0.93 S2)  ; 1, R-1-1-1
        (  -22.38   -23.78    -1.82     0.93 S2)  ; 2
        (  -23.39   -23.92    -2.52     0.93 S2)  ; 3
        (  -24.33   -23.66    -3.32     0.93 S2)  ; 4
        (  -23.91   -22.53    -3.42     0.93 S2)  ; 5
        (  -25.00   -20.96    -3.72     0.93 S2)  ; 6
        (  -25.50   -20.07    -3.82     0.93 S2)  ; 7
        (  -25.74   -18.87    -3.92     0.93 S2)  ; 8

[...]
```

These morphologies actually are reconstructed form a stack of up to 6 slices (to get the full axon projection) and I think that, in the notation `S2 "V2 section" -25 25 25`, `S2` is the identifier for this slice (and then it is used in segment identification   `(  -22.38   -23.78    -1.82     0.93 S2)  ; 2` to state that this segment is taken from this slice), the first number of ` -25 25 25` is the Z offset of this slice, and the second and third numbers somehow specifies the thickness (although I don't know the difference between these two numbers; they are always equal in our samples).

The Pneumatik library (which I think uses Neuron for display purpose...?) can load these files without issues... To load them with the current version of NeuroM, a working get-around is to delete 

```
(Sections S1 "V2 section" 0 25 25
 S2 "V2 section" -25 25 25
 S3 "V2 section" 25 25 25
 S4 "V2 section" 50 25 25
 S5 "V2 section" -50 25 25
) ; End of Sections
```

and every subsequent reference to these `SX` identifiers (i.e. replacing  `(  -22.38   -23.78    -1.82     0.93 S2)  ; 2` by  `(  -22.38   -23.78    -1.82     0.93)  ; 2` ).  The original files and the files modified that way are both rendered identically by Pneumatik, meaning that the slice offsets are already applied to the segment coordinates so that the `SX` mentions in these coordinate tuples is probably only useful at this time to be able to track the slice provenance of every segment for subsequent verifications and such... I guess that as a first step, NeuroM could be made compatible with these files using a similar approach (i.e., ignoring the Sections definition header and then ignoring the `SX` mentions in the coordinate tuples )... I'll let you see if you consider that the slice provenance is worth conserving and exposing in NeuroM!

**Comment by lidakanari** [comment]
If the SX data define the slice number are they relevant information that should be kept or can they just be ignored during the loading? (I think it would be very tricky to change the internal representation to add one more field, but maybe ignoring those fields could make sense, if your current solution is deleting them from the file).

**Comment by christian-oreilly** [comment]
Yeah, as I was trying to explain in the last part of my message, just ignoring them looks to work fine (at least, Pneumatik reconstruct both version (one original and one ignoring SX numbers) identically). As long as there is not a real use-case for having access to these slice numbers in NeuroM, it seems a reasonable option to just ignore these SX so that these files can be loaded without a lot of recoding efforts...

**Comment by mgeplf** [comment]
I'll try and fix this today or tomorrow, FYI.

**Comment by mgeplf** [comment]
fixed in #535

**Comment by mgeplf** [comment]
Please reopen this if it doesn't solve your problem.
